src/registry/date/validation-dateconstraint.stories.ts

```diff
@@ -0,0 +1,187 @@
+  component: DateConstraintValidation,
+  decorators: [withFormik],
+  parameters: {
+    modal: {noModal: true},
+    formik: {
+      initialValues: {
+        openForms: {
+          minDate: {mode: ''},
+          maxDate: {mode: ''},
+        },
+        datePicker: {
+          showWeeks: true,
+          startingDay: 0,
+          initDate: '',
+          minMode: 'day',
+          maxMode: 'year',
+          yearRows: 4,
+          yearColumns: 5,
+          minDate: null,
+          maxDate: null,
+        },
+      },
+    },
+  },
+  args: {
+    constraint: 'minDate',
+  },
+  argTypes: {
+    constraint: {
+      options: ['minDate', 'maxDate'],
+      control: {type: 'inline-radio'},
+    },
+  },
+} as Meta<typeof DateConstraintValidation>;
+
+type Story = StoryObj<typeof DateConstraintValidation>;
+
+export const InitialState: Story = {};
+
+export const FixedValue: Story = {
+  parameters: {
+    formik: {
+      initialValues: {
+        openForms: {
+          minDate: {
+            mode: 'fixedValue',
+          },
+          maxDate: {mode: ''},
+        },
+        datePicker: {
+          showWeeks: true,
+          startingDay: 0,
+          initDate: '',
+          minMode: 'day',
+          maxMode: 'year',
+          yearRows: 4,
+          yearColumns: 5,
+          minDate: '2023-01-01',
+          maxDate: null,
+        },
+      },
+    },
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    // Expand the panel
+    await userEvent.click(canvas.getByText(/date/));
+
+    expect(await canvas.findByText('Fixed value')).toBeVisible();
+    const datefield = await canvas.findByLabelText('Minimum date');
+    expect(datefield).toBeVisible();
+    expect(datefield).toHaveDisplayValue('2023-01-01');
+  },
+};
+
+export const FutureOrPastIncludingToday: Story = {
+  parameters: {
+    formik: {
+      initialValues: {
+        openForms: {
+          minDate: {
+            mode: 'future',
+            includeToday: true,
+          },
+          maxDate: {mode: ''},
+        },
+        datePicker: {
+          showWeeks: true,
+          startingDay: 0,
+          initDate: '',
+          minMode: 'day',
+          maxMode: 'year',
+          yearRows: 4,
+          yearColumns: 5,
+          minDate: '2023-01-01',
+          maxDate: null,
+        },
+      },
+    },
+  },
+
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    // Expand the panel
+    await userEvent.click(canvas.getByText(/date/));
+
+    expect(await canvas.findByText('In the future')).toBeVisible();
+    const checkbox = await canvas.findByLabelText('Including today');
+    expect(checkbox).toBeVisible();
+    expect(checkbox).toBeChecked();
+  },
+};
+
+export const RelativeToVariable: Story = {
+  parameters: {
+    formik: {
+      initialValues: {
+        openForms: {
+          minDate: {
+            mode: 'relativeToVariable',
+            variable: 'now',
```

Comment:
I was a bit confused by the lighter colour of the 'now' (`#495057`) with respect to 'Add' (`hsl(20,0,0)`, which is `#000000`).

![Screenshot from 2023-09-04 12-04-28](https://github.com/open-formulieren/formio-builder/assets/19154114/5a07eda3-f0cc-4629-a583-dfde35aae3a8)

Reply:
oh that is not deliberate at all, let me look into it!

Reply:
Figured it out - in the django admin they all have the same color (`#333` == `hsl(0, 0, 20)`) because the django styles modify the text input color. Discussed in the office that the pure bootstrap styles for storybook will remain as-is.